### PR TITLE
Feature/2429 test navigation improvements

### LINKED ISSUE #2427: Admin refresh - Consolidate screen space on Exercise pages
## Background
The Admin interface has grown organically over time and, in response to user feedback, there is a desire to group tasks, reports and other activities on the platform more intuitively to support both experienced and novice users. Additionally, this is an opportunity to refresh the display and make better use of screen space.

## User Story
As an Admin, I would like the tasks, reports and functions to be grouped more intuitively and logically on the platform so that I can quickly find what I am looking for.

## Benefit(s) to user (if not already clear from User story)
Make better use of screen real estate.
Dispense with unused functionality.

## Feature(s) Description

- [ ] Remove the `This is a new service...` line on Exercises pages
- [ ] Move Exercise reference to right hand end of navigation row

- [ ] Move `Create exercise` under `Archived exercises` in `Exercises` submenu
- [ ] Move `Copy to clipboard` to `Exercises` submenu and rename to `Copy exercise to clipboard` - it is currently a button - it should come after `Create exercise`
- [ ] Move `Archive exercise` to `Exercises` submenu - it is currently a button - it should come after `Copy exercise to clipboard`
- [ ] Move Add to favourite (and Remove from Favourite and Favourite label when they appear) under exercise title as below:

## Acceptance Criteria 
It's done when the above feature requests have been implemented.

## Definition of Done
- [ ] User stories/acceptance criteria met
- [ ] Internal reviews passed (feedback actioned)
- [ ] User testing passed
- [ ] Relevant technical documentation updated
- [ ] User guidance updated
- [ ] Deployed and merged without errors

## User Testing Steps
Internal testing only

## Feature Champion
Nick

### LINKED ISSUE #2429: SPIKE: Test improvements to navigation
## Background
As part of the Admin refresh, we'd like to experiment with bringing the left hand menus up to the top of the screen so that we can use the full width of the screen to display exercise content. We'd like to achieve a layout like this:

## Feature(s) Description
Experiment with the Exercise sub menu with the aim of delivering the above configuration. Ideally, we'd like to: 
1. Have the items in the sub-menu display in a grey box underneath the top navigation
2. Have the selected item in the sub menu appear next to the top level menu item separated by a >
3. Have a blue line delineating which menu items are selected
4. The menu that appears could either float on top of screen content or push the screen content down

### PR DESCRIPTION
**Pull request by drieJAC**
## What's included?

1. Have the items in the sub-menu display in a grey box underneath the top navigation (see Figure 1 below)
2. Have a grey line delineating which menu items are selected
3. The menu that appears pushes the screen content down
4. Remove archive button and put into top Exercises menu
5. Remove'copy to clipboard' button and put into top Exercises menu
6. Remove add/remove favourite button and put beneath exercise title

Closes #2429 
Closes #2427 

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
Go to: https://jac-admin-develop--pr2480-feature-2429-test-na-9n4vic5f.web.app/
and choose an exercise

1. Ensure the tab links and sub menu links work
2. Ensure the correct links are shown for exercises in different states.
3. Ensure the criteria in the following ticket are also met: https://github.com/jac-uk/admin/issues/2427

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Additional context
Figure 1

Screenshot 2024-07-24 at 18 17 07

## Related permissions
Have permissions been considered for this functionality?
- No permission changes required

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

**Comment by nickaddy** [comment]
@drieJAC Two questions:
- Is it possible to see this submenu in action?
- Are you content that we will be able to manage the dynamic tasks submenu?

**Comment by drieJAC** [comment]
@nickaddy I needed to fix something before the preview url was created. That's now fixed so you can use the preview url to view the submenu.
It would be good to check the scenarios under which the submenu items should appear. I expect there will need to be a few changes.

**Comment by warrensearle** [comment]
Looking good but spotted the following:

- [ ] The up/down arrows need to be reversed. Should be: down arrow (v) when menu is closed and can be opened; up arrow (^) when menu is open and can be closed
- [ ] Link colour should stay as it was (blue for links; black when current) rather than changing to purple (unless I've missed this request)

**Comment by nickaddy** [comment]
> Looking good but spotted the following:
> 
> * [ ]  The up/down arrows need to be reversed. Should be: down arrow (v) when menu is closed and can be opened; up arrow (^) when menu is open and can be closed
> * [ ]  Link colour should stay as it was (blue for links; black when current) rather than changing to purple (unless I've missed this request)

@drieJAC Agreed. Could you mirror this change on the Live Exercises page pls?

**Review by HalcyonJAC — CHANGES_REQUESTED** [review]
Code looks good to me.
I spotted something. If I click `Exercise` and then click `Applications`, the submenu will not disappear. Please see the following demo: 

https://github.com/jac-uk/admin/assets/79906532/dd03f824-fe86-47f4-ad7f-c93acb04f549

**Comment by nickaddy** [comment]
Good spot, @HalcyonJAC - looks like that happens on all menu items except Dashboard...

**Review by NormaJAC — COMMENTED** [review]
The tab links and sub-menu links work. The sub-menu arrows haven't changed position, still showing up arrow for closed. down arrow for open.

**Comment by nickaddy** [comment]
@drieJAC Love the new menu! I have a couple of minor requests:

- Pls increase the font size of the reference number to same size as the rest of the text on that row, it's a bit small at the moment, maybe you could make it bold too?
image
- Copy exercise to clipboard is not appearing in the Exercises menu - can you pls add this after Create exercise?

**Comment by drieJAC** [comment]
@nickaddy 
I've fixed the font-size of the reference number.
The 'copy to clipboard' will only appear if the exercise is approved (and the user has perm to update exercises).
Please confirm this is ok and I'll merge

File: src/views/Exercise/Details.vue [hunk 1]
```diff
@@ -1,98 +1,99 @@
...
+  //   },
+  // },
+  // watch: {
+  //   '$route': {
+  //     immediate: true,
+  //     handler(newRoute) {
+  //       this.$store.dispatch('exerciseCreateJourney/setCurrentRoute', newRoute.name);
+  //     },
+  //   },
+  // },
```
Comment (HalcyonJAC) [inline src/views/Exercise/Details.vue:97]: Should we remove the commented code?

File: src/views/Exercise.vue [hunk 2]
```diff
@@ -321,42 +383,42 @@ export default {
...
+    //     });
+    //   } else {
+    //     this.$store.dispatch('exerciseDocument/archive');
+    //     logEvent('info', 'Exercise archived', {
+    //       exerciseId: this.exerciseId,
+    //       exerciseRef: this.exercise.referenceNumber,
+    //     });
+    //   }
+    //   this.$refs.archiveModal.closeModal();
+    // },
```
Comment (HalcyonJAC) [inline src/views/Exercise.vue:421]: Should we remove the commented code?

File: src/views/Exercise.vue [hunk 3]
```diff
@@ -224,29 +259,29 @@ export default {
...
+    //     return 'By clicking accept you authorise the exercise to be archived';
+    //   }
+    // },
+    // archiveButtonText() {
+    //   if (this.isArchived) {
+    //     return 'Accept - unarchive this exercise';
+    //   } else {
+    //     return 'Accept - archive this exercise';
+    //   }
+    // },
```
Comment (HalcyonJAC) [inline src/views/Exercise.vue:284]: Should we remove the commented code?

**Review by HalcyonJAC — APPROVED** [review]
👍

**Review by KoWeiJAC — APPROVED** [review]

**Pull request merged**